### DISCUSSION TITLE
NX + eslint + typescript + Webstorm

### DISCUSSION THREAD

**Discussion by wanjas** [nrwl/nx#13683] [General]
Hello, 

Following is obviously more of a Webstorm issue but I wonder if somebody in NX community found a way to deal with this? 

NX creates `.eslintrc` config in every project which totally makes sense. But when it's combined with Typescript and Webstorm (or any other JetBrains  IDE) it leads to extreme memory consumption. Webstorm spawns eslint instances that are eating up to 1.9gb each when they try to lint typescript. 

One option is to get rid of `.eslintrc` in every project but I don't really want to do that. 
Another option I see is to buy more memory. 

A third one is to ask JetBrains for a solution like I've done here:
https://youtrack.jetbrains.com/issue/WEB-57163/Option-to-make-eslint-language-services-expirable-or-limited-in-number

**Comment by exsesx** [comment]
I stumbled upon this article, https://jakeginnivan.medium.com/using-typescript-project-references-in-nx-b3462b2fe6d4, which, I believe, aimed to improve the performance of TS in Nx mono repo. I've tried it. It works-ish, improving performance, but it also breaks stuff in some scenarios, so I wouldn't recommend it.

However, I'm bumping this discussion because the performance of my monorepo has become very tied to memory, and my 16 GB of RAM is not sufficient anymore, which bothers me. I'm optimizing it for now by using Neovim instead of RAM-hungry IDE/editors and trying to minimize memory consumption where possible. Still, I'd like my team to benefit from performance improvements. Some "global" configuration or something would be great.

Also, I've just realized this discussion was created on Dec 7, 2022 🥲
